### PR TITLE
Fixes DialogComponent

### PR DESCRIPTION
**Pull request by Kritarie**
Ensure DialogComponent is capable of rendering a dialog requested immediately in onShow of a Step

### CODE REVIEW

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt [hunk 1]
```diff
@@ -29,8 +29,11 @@ public class DialogComponent @Inject constructor() : LifecycleAware {
     createDialog()
   }
 
-  override fun resume(context: Context) {
+  override fun show(context: Context) {
```
Comment (Kritarie) [inline magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:32]: The meat of the change: move context assignment to `show` from `resume`, and context release to `hide` from `pause`.

File: magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt [hunk 2]
```diff
@@ -39,10 +42,14 @@ public class DialogComponent @Inject constructor() : LifecycleAware {
...
   override fun pause(context: Context) {
     destroyDialog()
+  }
+
+  override fun hide(context: Context) {
     this.context = null
   }
 
   private fun createDialog() {
+    shouldRestoreDialog = false
```
Comment (cmathew) [inline magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:52]: will this prevent dialogs from surviving multiple pause/resume runs? Hmm I do see `shouldRestoreDialog` gets set back to true ondestroy

Comment (Kritarie) [reply magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:52]: [device-2023-02-27-151404.webm](https://user-images.githubusercontent.com/5955888/221673517-09d8a2df-ef1e-43d1-af40-276ffb3245a3.webm)

Comment (Kritarie) [reply magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt:52]: The dialog in this example survives several pause/resume cycles, thanks to `shouldRestoreDialog`

**Review by kyeungk — APPROVED** [review]

**Pull request merged**